nixos/modules/programs/wivrn.nix

```diff
@@ -0,0 +1,43 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.wivrn;
+in
+{
+  options = {
+    programs.wivrn = {
+      enable = mkEnableOption (lib.mdDoc "Whether to enable WiVRn, an OpenXR streaming application.");
+
+      package = mkPackageOption pkgs "wivrn" { };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Whether to open the default ports in the firewall for the WiVRn server.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    services.avahi = {
+      enable = lib.mkDefault true;
+      publish = {
+        enable = lib.mkDefault true;
+        userServices = lib.mkDefault true;
+      };
+    };
```

Comment:
I assume these are required by WiVRn to work, so I think we should just enable them in general. Otherwise, we could also provide an assertion, so users know *why* these have to be enabled

Reply:
Yes, it does require Avahi. I just found that snippet from some other package so I will modify it.